runtime/dali/src/lib.rs

```diff
@@ -1673,21 +1677,31 @@ impl_runtime_apis! {
 			Transfer::get_denom_traces(key, offset, limit, count_total)
 		}
 
-		fn block_events() -> Vec<pallet_ibc::events::IbcEvent> {
-			let events = frame_system::Pallet::<Self>::read_events_no_consensus().into_iter().filter_map(|e| {
-				let frame_system::EventRecord{ event, ..} = *e;
-				match event {
-					Event::Ibc(pallet_ibc::Event::IbcEvents{ events }) => {
-						Some(events)
-					},
-					_ => None
-				}
-			});
-
-			events.fold(vec![], |mut events, ev| {
-				events.extend(ev);
-				events
-			})
+		fn block_events(extrinsic_index: Option<u32>) -> Option<Vec<pallet_ibc::events::IbcEvent>> {
```

Comment:
`Option<Vec<_>>` is a bad pattern, you can just return an empty vec.

Reply:
Yeah, that's how the index is gotten in frame-system, tested it already though and it works.